### PR TITLE
resolve #10

### LINKED ISSUE #10: Create Packages meta

### PR DESCRIPTION
**Pull request by Guseyn**

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#11](https://github.com/artipie/pypi-adapter/issues/11) is assigned to @iakunin[/z](https://www.0crat.com/u/iakunin), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/1e88922b-0525-4cc9-b830-1a39b2d074c2); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by g4s8** [comment]
@0crat assign me

File: pom.xml [hunk 1]
```diff
@@ -1,154 +1,120 @@
...
-                                <exclude>checkstyle:/src/test/resources/packaging_tutorial/example_pkg/server/.*</exclude>
-                            </excludes>
-                        </configuration>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-    </profiles>
+    <artifactId>ppom</artifactId>
+    <version>0.1.3</version>
```
Comment (g4s8) [inline pom.xml]: @Guseyn let's use latest `0.2.0` version here

File: pom.xml [hunk 2]
```diff
@@ -1,154 +1,120 @@
...
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+  <artifactId>pypi-adapter</artifactId>
+  <version>1.0-SNAPSHOT</version>
+  <name>pypi-adapter</name>
+  <url>https://github.com/artipie/pypi-adapter</url>
+  <properties>
```
Comment (g4s8) [inline pom.xml]: @Guseyn these properties are not needed anymore

File: pom.xml [hunk 3]
```diff
@@ -1,154 +1,120 @@
...
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <maven.compiler.source>11</maven.compiler.source>
+    <maven.compiler.target>11</maven.compiler.target>
+    <junit-platform.version>5.4.2</junit-platform.version>
+  </properties>
+  <licenses>
+    <license>
+      <name>MIT License</name>
+      <url>https://github.com/artipie/pypi-adapter/blob/master/LICENSE</url>
```
Comment (g4s8) [inline pom.xml]: @Guseyn license should be named `LICENSE.txt`

File: pom.xml [hunk 4]
```diff
@@ -1,154 +1,120 @@
...
+    </license>
+  </licenses>
+  <distributionManagement>
```
Comment (g4s8) [inline pom.xml]: @Guseyn `distributionManagement` is already defined in parent pom, please remove it here

File: pom.xml [hunk 5]
```diff
@@ -1,154 +1,120 @@
...
+          <artifactId>jcip-annotations</artifactId>
+        </exclusion>
+        <exclusion>
+          <groupId>com.google.code.findbugs</groupId>
+          <artifactId>jsr305</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <!-- Test -->
+    <dependency>
```
Comment (g4s8) [inline pom.xml]: @Guseyn why don't you use junit-5?

File: src/main/java/com/artipie/pypi/ArtifactMeta.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    /**
+     * Ctor.
+     *
+     * @param name Name of artifact.
+     * @param version Version of artifact.
+     */
+    public ArtifactMeta(final String name, final String version) {
+        this.name = name;
+        this.version = version;
+        this.pattern = "<tr><td><a href=\"%s-%s.tar.gz\">%s-%s.tar.gz</a></td></tr>";
```
Comment (g4s8) [inline src/main/java/com/artipie/pypi/ArtifactMeta.java]: @Guseyn you either need to add `pattern` constructor parameter or move it to `private static` field

File: src/main/java/com/artipie/pypi/PackageMeta.java [hunk 7]
```diff
@@ -49,16 +44,14 @@
...
      *
      * @param name Name of Package.
-     * @param version Version of Package.
      */
-    public PackageMeta(final String name, final String version) {
+    public PackageMeta(final String name) {
         this.name = name;
-        this.version = version;
-        this.pattern = "<tr><td><a href=\"%s-%s.tar.gz\">%s-%s.tar.gz</a></td></tr>";
+        this.pattern = "<tr><td><a href=\"%s\">%s</a></td></tr>";
```
Comment (g4s8) [inline src/main/java/com/artipie/pypi/PackageMeta.java]: @Guseyn same here: you can't inject this pattern, and it's always the same

**Comment by 0crat** [comment]
> @0crat assign me ([here](https://github.com/artipie/pypi-adapter/issues/11#issuecomment-582808910))

@g4s8 There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 12@6189cca5e2fd, thread: PQ-CT2E6TK9B
com.zerocracy.claims.ClaimIn[214] java.lang.IllegalArgumentException: Parameter "age" not found in "Order was canceled" among: [comment, job, issue, repo, cause, _parent_sig, flow, login]
```

[0.51.1](https://github.com/zerocracy/farm): CID: [f3cedbfd-4d32-4300-a735-90ee8fd153f1](https://www.0crat.com/footprint/CT2E6TK9B/f3cedbfd-4d32-4300-a735-90ee8fd153f1), Type: "Order was canceled"

**Comment by 0crat** [comment]
> @0crat assign me ([here](https://github.com/artipie/pypi-adapter/issues/11#issuecomment-582808910))

@g4s8 This pull request [#11](https://github.com/artipie/pypi-adapter/issues/11) is assigned to @g4s8[/z](https://www.0crat.com/u/g4s8), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/92b84451-67d1-49ff-b28f-4b435427b3ab); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by Guseyn** [comment]
@g4s8 please check the updates

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by yegor256** [comment]
@Guseyn thanks!

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) resigned from [#11](https://github.com/artipie/pypi-adapter/issues/11), since the job is not in scope anymore

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 11@706b9e63c3bc, thread: PQ-CT2E6TK9B
com.zerocracy.claims.ClaimIn[214] java.lang.IllegalArgumentException: Parameter "age" not found in "Order was canceled" among: [reason, priority, job, cause, _parent_sig, flow, voluntarily, login]
```

[0.51.6](https://github.com/zerocracy/farm): CID: [cff7b3cb-4443-4a76-aa27-5180297c9ef5](https://www.0crat.com/footprint/CT2E6TK9B/cff7b3cb-4443-4a76-aa27-5180297c9ef5), Type: "Order was canceled"

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Pull request merged**